parts/windows/csecmd.ps1

```diff
@@ -1,22 +1,19 @@
-# This csecmd is not used until servicePrincipalClientSecret is encoded,
-# keeping this file for now to further facilicate the future removing ARM Template
 echo %DATE%,%TIME%,%COMPUTERNAME% && powershell.exe -ExecutionPolicy Unrestricted -command \"
 $arguments = '
--MasterIP {{ GetKubernetesEndpoint }} 
--KubeDnsServiceIp {{ GetParameter "kubeDNSServiceIP" }} 
--MasterFQDNPrefix {{ GetParameter "masterEndpointDNSNamePrefix" }} 
--Location {{ GetVariable "location" }} 
+-MasterIP {{ GetKubernetesEndpoint }}
+-KubeDnsServiceIp {{ GetParameter "kubeDNSServiceIP" }}
+-MasterFQDNPrefix {{ GetParameter "masterEndpointDNSNamePrefix" }}
+-Location {{ GetVariable "location" }}
 {{if UserAssignedIDEnabled}}
--UserAssignedClientID {{ GetVariable "userAssignedIdentityID" }} 
+-UserAssignedClientID {{ GetVariable "userAssignedIdentityID" }}
 {{ end }}
--TargetEnvironment {{ GetTargetEnvironment }} 
--AgentKey {{ GetParameter "clientPrivateKey" }} 
--AADClientId {{ GetParameter "servicePrincipalClientId" }} 
--AADClientSecret ''{{ GetParameter "servicePrincipalClientSecret" }}''
--NetworkAPIVersion 2018-08-01;
-$inputFile = '%SYSTEMDRIVE%\AzureData\CustomData.bin'; 
+-TargetEnvironment {{ GetTargetEnvironment }}
+-AgentKey {{ GetParameter "clientPrivateKey" }}
+-AADClientId {{ GetParameter "servicePrincipalClientId" }}
+-AADClientSecret ''{{ GetParameter "encodedServicePrincipalClientSecret" }}''
```

Comment:
The double single quote is first introduced in https://github.com/Azure/aks-engine/pull/927, based64 encoded secret only may also work.  But for now, we just keep it as it is for safety.

Reply:
Will agentbaker to update this file to replace the the template variables with the real value?
For example, 
-AADClientSecret ''{{ GetParameter "encodedServicePrincipalClientSecret" }}''
=>
-AADClientSecret XXX-SECRET

Reply:
No, we can only obtained base64-encode secrets here,  and the real value will be decoded [when writing azure.json](https://github.com/Azure/AgentBaker/blob/master/parts/windows/kuberneteswindowssetup.ps1#L337)

for example `-AADClientSecret ''U2VjcmV0''` in 
`echo %DATE%,%TIME%,%COMPUTERNAME% && powershell.exe -ExecutionPolicy Unrestricted -command " $arguments = ' -MasterIP uttestdom-dns-5d7c849e.hcp.southcentralus.azmk8s.io -KubeDnsServiceIp 10.0.0.10 -MasterFQDNPrefix uttestdom -Location southcentralus  -TargetEnvironment AzurePublicCloud -AgentKey  -AADClientId ClientID -AADClientSecret ''U2VjcmV0'' -NetworkAPIVersion 2018-08-01'; $inputFile = '%SYSTEMDRIVE%\AzureData\CustomData.bin'; $outputFile = '%SYSTEMDRIVE%\AzureData\CustomDataSetupScript.ps1'; Copy-Item $inputFile $outputFile; Invoke-Expression('{0} {1}' -f $outputFile, $arguments); " > %SYSTEMDRIVE%\AzureData\CustomDataSetupScript.log 2>&1; exit $LASTEXITCODE `

Reply:
I see. It should be ok since the template variables will be replaced by the real value. For secret, it is encoded value.